renku/core/management/datasets.py

```diff
@@ -110,25 +110,36 @@ def load_dataset(self, name=None):
     def with_dataset(self, name=None):
         """Yield an editable metadata object for a dataset."""
         dataset = self.load_dataset(name=name)
+        clean_up_required = False
```

Comment:
Since there are several different possible cleanups that we could do in case of failure, I wouldn't do it here. Broken references are just one case in which we have to clean up, but I've seen others that could happen (e.g. Uncleaned lock files, partially created metadata to name a few). What I'm thinking is we should come up with a general enough solution which can handle those problems which can happen along the chain. Maybe something similar like migrations?

Reply:
Let's discuss this tomorrow in Bern.

Reply:
As we discussed, I've created https://github.com/SwissDataScienceCenter/renku-python/issues/736.

Reply:
I'll add a TODO for it.

Reply:
Let's just add the ticket number to code with a comment so we know in the future.

Reply:
Let's add that ticket as a comment here as well so we know that we need to cleanup with that ticket.